common/src/main/java/com/wynntils/features/redirects/ChatRedirectFeature.java

```diff
@@ -503,10 +505,13 @@ protected StyledText getNotification(Matcher matcher) {
     }
 
     private class LoginRedirector extends SimpleRedirector {
+        // Test suite: https://regexr.com/7jhti
+        private static final String RANK_STRING =
+                Arrays.stream(PlayerRank.values()).map(PlayerRank::getTag).collect(Collectors.joining());
         private static final Pattern FOREGROUND_PATTERN = Pattern.compile(
-                "^§.\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §.(?<name>.*)§. has just logged in!$");
-        private static final Pattern BACKGROUND_PATTERN = Pattern.compile(
-                "^(?:§8)?\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §7(?<name>.*)§8 has just logged in!$");
+                "^(?<rank>[" + RANK_STRING + "]) §#[0-9a-f]{8}(?<name>[\\w ]{1,20})§[0-9a-f] has just logged in!$");
+        private static final Pattern BACKGROUND_PATTERN = Pattern.compile("^(?:§8)?\\[(§.)+\\|?(§.)*(?<rank>["
+                + RANK_STRING + "]) §#[0-9a-f]{8}(?<name>[\\w ]{1,20})§[0-9a-f] has just logged in!$");
```

Comment:
`§#[0-9a-f]{6, 8}` because we don't include the alpha every case (this might work but 6, 8 is a bit more comaptible)